src/components/Arguments/index.tsx

```diff
@@ -0,0 +1,254 @@
+import React, {useRef, useState} from "react";
+import {FaRegCheckCircle, FaRegTimesCircle, FaSpinner} from "react-icons/fa";
+import {motion} from "framer-motion";
+import {EntityType} from "providers/Project";
+import {useProject} from "providers/Project/projectHooks";
+import {Account, ResultType, useSetExecutionResultsMutation} from "api/apollo/generated/graphql";
+import {ControlContainer, HoverPanel, StatusMessage} from "./styles";
+import {Argument} from "./types";
+import {ActionButton, ArgumentsList, ArgumentsTitle, Signers} from "./components";
+
+type ArgumentsProps = {
+  type: EntityType,
+  list: Argument[],
+  signers: number,
+  validCode: boolean
+}
+
+const validateByType = (value: any, type: string) => {
+  if (value.length === 0){
+    return "Value can't be empty"
+  }
+
+  switch (true){
+    // Strings
+    case type === "String":{
+      return null // no need to validate String for now
+    }
+
+    // Integers
+    case type.includes("Int"):{
+      if (isNaN(value)){
+        return "Should be a valid Integer number"
+      }
+      return null
+    }
+
+    // Words
+    case type.includes("Word"):{
+      if (isNaN(value)){
+        return "Should be a valid Word number"
+      }
+      return null
+    }
+
+    // Fixed Point
+    case type.includes("Fix"):{
+      if (isNaN(value)){
+        return "Should be a valid fixed point number"
+      }
+      return null
+    }
+
+    // Address
+    case type === "Address":{
+      if (!value.match(/(^0x[\w\d]{16})|(^0x[\w\d]{1,4})/)){

```

Comment:
I wonder if we should make the prefix configurable?I think they are thinking about changing it.

Reply:
I think we can be retroactive in this specific case.
Should be easy to fix and for now it's not gonna change is close future 🤷‍♂️